zulipterminal/ui_tools/boxes.py

```diff
@@ -113,14 +113,17 @@ def autocomplete_mentions(self, text: str, state: int, prefix_string: str,
                               num_suggestions: int) -> Optional[str]:
         # Handles user mentions (@ mentions and silent mentions)
         # and group mentions.
-        group_typeahead = ['@*{}*'.format(group_name)
-                           for group_name in self.model.user_group_names
-                           if match_groups(group_name, text[1:])]
-
-        users_list = self.view.users
-        user_typeahead = [prefix_string+'**{}**'.format(user['full_name'])
-                          for user in users_list
-                          if match_user(user, text[len(prefix_string):])]
+        group_typeahead = []
+        for group_name in self.model.user_group_names:
+            if match_groups(group_name, text[1:]):
+                group_typeahead.append('@*{}*'.format(group_name))
```

Comment:
I'm wary of this refactoring. While I'm not averse to for loops, I think this could be simpler by constructing a filtered name list, and then a 'styled' list for the typeahead.

See main review note too.